src/client.ts

```diff
@@ -789,36 +796,72 @@ export class McpdClient {
   async getAgentTools(options?: {
     format?: "array";
     servers?: string[];
+    tools?: string[];
   }): Promise<AgentFunction[]>;
   async getAgentTools(options: {
     format: "object";
     servers?: string[];
+    tools?: string[];
   }): Promise<Record<string, AgentFunction>>;
   async getAgentTools(options: {
     format: "map";
     servers?: string[];
+    tools?: string[];
   }): Promise<Map<string, AgentFunction>>;
   async getAgentTools(
     options: AgentToolsOptions = {},
   ): Promise<
     AgentFunction[] | Record<string, AgentFunction> | Map<string, AgentFunction>
   > {
-    const { servers, format = "array" } = options;
+    const { servers, tools, format = "array" } = options;
+
+    const allTools = await this.agentTools(servers);
 
-    // Get tools in array format (default internal representation)
-    const tools = await this.agentTools(servers);
+    const filteredTools = tools
+      ? allTools.filter((tool) => this.#matchesToolFilter(tool, tools))
+      : allTools;
 
-    // Return in requested format
     switch (format) {
       case "object":
-        return Object.fromEntries(tools.map((tool) => [tool.name, tool]));
+        return Object.fromEntries(
+          filteredTools.map((tool) => [tool.name, tool]),
+        );
 
       case "map":
-        return new Map(tools.map((tool) => [tool.name, tool]));
+        return new Map(filteredTools.map((tool) => [tool.name, tool]));
 
       case "array":
       default:
-        return tools;
+        return filteredTools;
     }
   }
+
+  /**
+   * Check if a tool matches the tool filter.
+   *
+   * Supports two formats:
+   * - Raw tool name: "get_current_time" (matches tool name only)
+   * - Server-prefixed name: "time__get_current_time" (server + TOOL_SEPARATOR + tool)
+   *
+   * @param tool The tool to check.
+   * @param tools List of tool names/patterns to match against.
+   * @returns True if the tool matches any item in the filter.
+   */
+  #matchesToolFilter(tool: AgentFunction, tools: string[]): boolean {
+    return tools.some((filterItem) => {
+      const separatorIndex = filterItem.indexOf(TOOL_SEPARATOR);
+      if (separatorIndex !== -1) {
+        const leftPart = filterItem.slice(0, separatorIndex).trim();
+        const rightPart = filterItem
+          .slice(separatorIndex + TOOL_SEPARATOR.length)
+          .trim();
+
+        if (leftPart && rightPart && filterItem === tool.name) {
+          return true;
+        }
+      }
+
+      return filterItem === tool._toolName;
+    });
```

Comment:
would something like this work?


```suggestion
  return tools.some((toolFilter) => {
    if (toolFilter.includes(TOOL_SEPARATOR)) {
      return toolFilter.trim() === tool.name;
    } else {
      return toolFilter === tool._toolName;
    }
  });
```

Reply:
Will do something like this (thanks!) the current logic I added was supposed to be to show that we validated the left/right bits, but TBH it's not super required.